api/src/core/Directus/Database/TableGateway/RelationalTableGateway.php

```diff
@@ -1138,6 +1133,7 @@ public function fetchItems(array $params = [], \Closure $queryCallback = null)
             }
         }
 
+        $selectedFields = $countOnly ? ['total' => new Expression('COUNT(*)')] : $selectedFields;
```

Comment:
when pagination is used in request, there will be an 'offset' & 'limit' params, so, sql query like 'select count(*) from table limit n offset x' (x > 0) will return empty set, so you should unset it like:

```php
if ($countOnly) {
            $selectedFields = ['total' => new Expression('COUNT(*)')];
            unset($params['offset']);
            unset($params['limit']);
            unset($params['sort']);
        }
```